homeassistant/components/binary_sensor/wemo.py

```diff
@@ -41,48 +44,93 @@ def __init__(self, hass, device):
         """Initialize the WeMo sensor."""
         self.wemo = device
         self._state = None
+        self._available = True
+        self._update_lock = None
+
+        # look up model name, name, and serial number
+        # once as it incurs network traffic
+        self._model_name = self.wemo.model_name
+        self._name = self.wemo.name
+        self._serialnumber = self.wemo.serialnumber
```

Comment:
Just as general advice: It's not good/unexpected to do IO in a constructor - for example a bug in the chromecast integration that took weeks to resolve was due to unexpected work in the constructor.

Reply:
I can move these to async_added_to_hass, but serial number is used by the unique_id property. Is that going to cause a race condition with HASS wanting to get the unique_id for the entity registry?

Reply:
Ah crap ok. Yes `unique_id` is called before `async_added_to_hass` so that won't work.